pkg/tsdb/cloudwatch/get_metric_query_batches.go

```diff
@@ -0,0 +1,83 @@
+package cloudwatch
+
+import (
+	"regexp"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models"
+)
+
+// nonWordRegex is for spliting the expressions to just functions and ids
+var nonWordRegex = regexp.MustCompile(`\W+`)
+
+// getMetricQueryBatches separates queries into batches if necessary. Metric Insight queries cannot run together, and math expressions must be run
+// with all the queries they reference.
+func getMetricQueryBatches(queries []*models.CloudWatchQuery, logger log.Logger) [][]*models.CloudWatchQuery {
```

Comment:
(noticed that `logger` and `idToIndex` below are unused)

Reply:
fixed